### PR TITLE
rocAL - Tf pets training

### PR DESCRIPTION
**Pull request by swetha097**
This fixes the tf pets training to run in the latest pytorch & tf dockers

### CODE REVIEW

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# installing required packages
+pip install contextlib2
+git clone https://github.com/tensorflow/models.git
+cd models/research
+protoc object_detection/protos/*.proto --python_out=.
+cp object_detection/packages/tf2/setup.py .
+python -m pip install --use-feature=2020-resolver .
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:7]: @swetha097 The flag "2020-resolver" was not working when I tried previously. The install worked with removing the flag altogether. ``` option use-feature: invalid choice: '2020-resolver'```
Is that not the situation anymore?
Also, if you notice all these are installed in the [tensorflow docker](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/0a4e22ab837ea4e0efcf215fe1d99cae36c6df7b/docker/tensorflow/mivisionx-with-tensorflow.dockerfile#L55). Why do we need a separate script again?

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh [hunk 2]
```diff
@@ -14,8 +13,10 @@ tar xzvf images.tar.gz
 tar xzvf annotations.tar.gz
 mkdir tf_pets_records
 git clone https://github.com/swetha097/rocALmodels.git
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh:15]: @swetha097  Where is this repo originally coming from? We might need to add it into MIVisionX-data or someplace common if no copyright issues

**Review by LakshmiKumar23 — CHANGES_REQUESTED** [review]

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh (hunk 1 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:7]: Yes Lakshmi, we dont see a issue with this now in pytorch docker - `option use-feature: invalid choice: '2020-resolver'`

We have mentioned in the README that the requirements is only for pytorch docker & not for tensorflow docker

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh (hunk 2 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh:15]: I will check on this & get back to you Lakshmi

Comment (swetha097) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh:15]: Few files like https://github.com/swetha097/rocALmodels/blob/master/models/research/configs/ssd320_bench.config seems to have Nvidia copyright Lakshmi.
Few of them seem to have tensorflow copyright too. 

This codebase seems to originally come from tensorflow & has been modified to run on Nvidia.

Confirmed with Abishek on this as I had forked from his repository.

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh (hunk 1 above)
Comment (LakshmiKumar23) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:7]: @swetha097 Does the tf docker need the flag or not? If it doesn't please change the dockerfile also

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh (hunk 2 above)
Comment (LakshmiKumar23) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/download_and_preprocess_dataset.sh:15]: @kiritigowda How do we want to host this model for training example?

File: rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh (hunk 1 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:7]:
```
RUN cd ~/models/research/ && \
protoc object_detection/protos/*.proto --python_out=. && \
cp object_detection/packages/tf2/setup.py . && \
pip3 install --use-feature=2020-resolver .
```

This is the CMD used in tf docker, its same as what we gave for pytorch here.

Comment (swetha097) [reply rocAL/rocAL_pybind/examples/tf_petsTrainingExample/requirements.sh:7]: This flag is used there also.

**Comment by LakshmiKumar23** [comment]
@swetha097 I am unable to test this PR. I am seeing errors while running ./run.sh inside the tensorflow docker
```
In file included from /dockerx/test/MIVisionX/rocAL/rocAL_pybind/rocal_pybind.cpp:23:
/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = void* (*&)(void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int>, unsigned int, unsigned int, RocalResizeInterpolationType); Return = void*; Args = {void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int, std::allocator<unsigned int> >, unsigned int, unsigned int, RocalResizeInterpolationType}; Extra = {pybind11::name, pybind11::scope, pybind11::sibling, char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’:
/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:64:9:   required from ‘pybind11::cpp_function::cpp_function(Return (*)(Args ...), const Extra& ...) [with Return = void*; Args = {void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int, std::allocator<unsigned int> >, unsigned int, unsigned int, RocalResizeInterpolationType}; Extra = {pybind11::name, pybind11::scope, pybind11::sibling, char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’
/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:819:22:   required from ‘pybind11::module& pybind11::module::def(const char*, Func&&, const Extra& ...) [with Func = void* (*)(void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int>, unsigned int, unsigned int, RocalResizeInterpolationType); Extra = {char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’
/dockerx/test/MIVisionX/rocAL/rocAL_pybind/rocal_pybind.cpp:441:51:   required from here
**/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:133:50: error: static assertion failed: The number of argument annotations does not match the number of function arguments
  133 |         static_assert(expected_num_args<Extra...>(sizeof...(Args), cast_in::has_args, cast_in::has_kwargs),
      |**                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/rocal_pybind.dir/build.make:76: CMakeFiles/rocal_pybind.dir/rocal_pybind.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/rocal_pybind.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
Traceback (most recent call last):
  File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 77, in <module>
    setup(name='amd-rocal',
  File "/root/.local/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/root/.local/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 299, in run
    self.run_command('build')
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/command/build.py", line 132, in run
    self.run_command(cmd_name)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 52, in run
    self.build_extension(ext)
  File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 75, in build_extension
    subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
  File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j2']' returned non-zero exit status 2.
Going to install ./dist/*
ERROR: Invalid requirement: './dist/*'
Hint: It looks like a path. File './dist/*' does not exist.
ERROR: Invalid requirement: './dist/*'
Hint: It looks like a path. File './dist/*' does not exist.

[notice] A new release of pip available: 22.3.1 -> 23.0
[notice] To update, run: python3.9 -m pip install --upgrade pip
```

**Comment by swetha097** [comment]
> python3.9 -m pip install --upgrade pip



> @swetha097 I am unable to test this PR. I am seeing errors while running ./run.sh inside the tensorflow docker
> 
> ```
> In file included from /dockerx/test/MIVisionX/rocAL/rocAL_pybind/rocal_pybind.cpp:23:
> /dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h: In instantiation of ‘void pybind11::cpp_function::initialize(Func&&, Return (*)(Args ...), const Extra& ...) [with Func = void* (*&)(void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int>, unsigned int, unsigned int, RocalResizeInterpolationType); Return = void*; Args = {void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int, std::allocator<unsigned int> >, unsigned int, unsigned int, RocalResizeInterpolationType}; Extra = {pybind11::name, pybind11::scope, pybind11::sibling, char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’:
> /dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:64:9:   required from ‘pybind11::cpp_function::cpp_function(Return (*)(Args ...), const Extra& ...) [with Return = void*; Args = {void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int, std::allocator<unsigned int> >, unsigned int, unsigned int, RocalResizeInterpolationType}; Extra = {pybind11::name, pybind11::scope, pybind11::sibling, char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’
> /dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:819:22:   required from ‘pybind11::module& pybind11::module::def(const char*, Func&&, const Extra& ...) [with Func = void* (*)(void*, const char*, RocalImageColor, RocalDecodeDevice, unsigned int, unsigned int, unsigned int, unsigned int, bool, bool, bool, unsigned int, unsigned int, bool, RocalResizeScalingMode, std::vector<unsigned int>, unsigned int, unsigned int, RocalResizeInterpolationType); Extra = {char [158], pybind11::return_value_policy, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg_v, pybind11::arg, pybind11::arg, pybind11::arg_v}]’
> /dockerx/test/MIVisionX/rocAL/rocAL_pybind/rocal_pybind.cpp:441:51:   required from here
> **/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:133:50: error: static assertion failed: The number of argument annotations does not match the number of function arguments
>   133 |         static_assert(expected_num_args<Extra...>(sizeof...(Args), cast_in::has_args, cast_in::has_kwargs),
>       |**                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> make[2]: *** [CMakeFiles/rocal_pybind.dir/build.make:76: CMakeFiles/rocal_pybind.dir/rocal_pybind.cpp.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/rocal_pybind.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2
> Traceback (most recent call last):
>   File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 77, in <module>
>     setup(name='amd-rocal',
>   File "/root/.local/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
>     return distutils.core.setup(**attrs)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
>     return run_commands(dist)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
>     dist.run_commands()
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
>     self.run_command(cmd)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
>     super().run_command(command)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>     cmd_obj.run()
>   File "/root/.local/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 299, in run
>     self.run_command('build')
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
>     self.distribution.run_command(command)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
>     super().run_command(command)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>     cmd_obj.run()
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/command/build.py", line 132, in run
>     self.run_command(cmd_name)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
>     self.distribution.run_command(command)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
>     super().run_command(command)
>   File "/root/.local/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>     cmd_obj.run()
>   File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 52, in run
>     self.build_extension(ext)
>   File "/dockerx/test/MIVisionX/rocAL/rocAL_pybind/setup.py", line 75, in build_extension
>     subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
>   File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j2']' returned non-zero exit status 2.
> Going to install ./dist/*
> ERROR: Invalid requirement: './dist/*'
> Hint: It looks like a path. File './dist/*' does not exist.
> ERROR: Invalid requirement: './dist/*'
> Hint: It looks like a path. File './dist/*' does not exist.
> 
> [notice] A new release of pip available: 22.3.1 -> 23.0
> [notice] To update, run: python3.9 -m pip install --upgrade pip
> ```

We need to upgrade the pip here Lakshmi
python3.9 -m pip install --upgrade pip

**Comment by LakshmiKumar23** [comment]
> */dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:133:50: error: static assertion failed: The number of argument annotations does not match the number of function arguments
>   133 |         static_assert(expected_num_args<Extra...>(sizeof...(Args), cast_in::has_args, cast_in::has_kwargs),
>       |**                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> make[2]: *** [CMakeFiles/rocal_pybind.dir/build.make:76: CMakeFiles/rocal_pybind.dir/rocal_pybind.cpp.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/rocal_pybind.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2

@swetha097 That doesn't help. Please see the assertion error
```
*/dockerx/test/MIVisionX/rocAL/rocAL_pybind/third_party_lib/pybind11/include/pybind11/pybind11.h:133:50: error: static assertion failed: The number of argument annotations does not match the number of function arguments
  133 |         static_assert(expected_num_args<Extra...>(sizeof...(Args), cast_in::has_args, cast_in::has_kwargs),
      |**                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/rocal_pybind.dir/build.make:76: CMakeFiles/rocal_pybind.dir/rocal_pybind.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/rocal_pybind.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Review by LakshmiKumar23 — APPROVED** [review]
@swetha097 @kiritigowda This PR can be merged. A separate PR will be created to fix the Dockerfile issues

**Pull request merged**